include/xgboost/c_api.h

```diff
@@ -72,10 +63,17 @@ typedef float bst_float;  // NOLINT(*)
 //#endif
 
 
+#ifdef __SGX__
+/*! \brief handle to DMatrix */
+typedef char* DMatrixHandle;  // NOLINT(*)
```

Comment:
Is it confusing if we give the new type the same name? Especially to someone new navigating the codebase

If we want a different name, maybe something like `DMatrixAlias` would work?

Reply:
This is a good idea. Perhaps we can fix this later once the other features are merged in to the main branch? (It will potentially lead to numerous merge conflicts otherwise.)

Reply:
I'll open an issue for this so we don't forget.